pennylane/math/multi_dispatch.py

```diff
@@ -537,6 +537,11 @@ def einsum(indices, *operands, like=None):
     if like is None:
         like = get_interface(*operands)
     operands = np.coerce(operands, like=like)
+
+    if optimize is not None and like != "torch":
+        # torch einsum doesn't support the optimize keyword argument
+        return np.einsum(indices, *operands, like=like, optimize=optimize)
+
     return np.einsum(indices, *operands, like=like)

```

Comment:
Might be easier to read if we flip the logic. Controlling on the the negative forces the brain to jump through an extra hoop.

```suggestion
    if optimize is None or like == "torch":
        # torch einsum doesn't support the optimize keyword argument
        return np.einsum(indices, *operands, like=like)
    return np.einsum(indices, *operands, like=like, optimize=optimize)
```